src/main/scala/io/kaizensolutions/virgil/CQLExecutor.scala

```diff
@@ -0,0 +1,215 @@
+package io.kaizensolutions.virgil
+
+import com.datastax.oss.driver.api.core.cql.{BatchType => _, _}
+import com.datastax.oss.driver.api.core.{CqlSession, CqlSessionBuilder}
+import io.kaizensolutions.virgil.configuration.{ExecutionAttributes, PageState}
+import io.kaizensolutions.virgil.renderer.CqlStatement
+import io.kaizensolutions.virgil.Proofs._
+import zio._
+import zio.stream._
+
+import scala.jdk.CollectionConverters._
+
+/**
+ * CQLExecutor is a ZIO based wrapper for the Apache Cassandra Java Driver that
+ * provides ZIO and ZStream abstractions over the Datastax Java driver. We
+ * consider CQLExecutor to be the interpreter of [[CQL[A]] expressions.
+ *
+ * @param underlyingSession
+ *   is the underlying Datastax Java driver session
+ */
+class CQLExecutor(underlyingSession: CqlSession) {
```

Comment:
Consider making this a trait, so there could exist a test implementation.

Also consider convenience methods on `CQL`, such that one could do, `cql.execute` and end up with this trait inside the environment.

Reply:
Done and done